python/dgl/mock_sparse/reduce.py

```diff
@@ -0,0 +1,286 @@
+    >>> A = SparseMatrix(row, col, val, shape=(4, 3))
+    >>> print(A.sum())
+    tensor([5, 5])
+    >>> print(A.sum(0))
+    tensor([[3, 3], [0, 0], [2, 2]])
+    """
+    return A.reduce(dim, rtype='sum')
+
+
+def smax(A: SparseMatrix, dim=None):
+    """Returns the maximum of sparse matrix A on dimension `dim`.
+
+    If `dim` is None, it reduces all the elements in the sparse matrix. Otherwise,
+    it reduces on the row (dim=0) or column (dim=1) dimension, producing a tensor
+    of shape `(A.shape[1], ) + A.val.shape[:1]` or `(A.shape[0],) + A.val.shape[:1]`.
+
+    The reduction does not count zero values. If the row or column to be reduced
+    does not have any non-zero value, the result will be 0.
+
+    Parameters
+    ----------
+    dim : int, optional
+        The dimension to reduce.
+
+    Returns
+    ----------
+    Tensor
+        Reduced tensor
+
+    Examples
+    ----------
+
+    >>> row = torch.tensor([0, 1, 1])
+    >>> col = torch.tensor([0, 0, 2])
+    >>> val = torch.tensor([1, 1, 2])
+    >>> A = SparseMatrix(row, col, val, shape=(4, 3))
+    >>> print(A.smax())
+    tensor(2)
+    >>> print(A.smax(0))
+    tensor([1, 0, 2])
+    >>> print(A.smax(1))
+    tensor([1, 2, 0, 0])
+
+    >>> row = torch.tensor([0, 1, 1])
+    >>> col = torch.tensor([0, 0, 2])
+    >>> val = torch.tensor([[1, 2], [2, 1], [2, 2]])
+    >>> A = SparseMatrix(row, col, val, shape=(4, 3))
+    >>> print(A.max())
+    tensor([2, 2])
+    >>> print(A.max(0))
+    tensor([[2, 2], [0, 0], [2, 2]])
+    >>> print(A.max(1))
+    tensor([[1, 2], [2, 2], [0, 0], [0, 0]])
+    """
+    return A.reduce(dim, rtype='smax')
+
+
+def smin(A: SparseMatrix, dim=None):
+    """Returns the minimum of sparse matrix A on dimension `dim`.
+
+    If `dim` is None, it reduces all the elements in the sparse matrix. Otherwise,
+    it reduces on the row (dim=0) or column (dim=1) dimension, producing a tensor
+    of shape `(A.shape[1], ) + A.val.shape[:1]` or `(A.shape[0],) + A.val.shape[:1]`.
+
+    The reduction does not count zero values. If the row or column to be reduced
+    does not have any non-zero value, the result will be 0.
+
+    Parameters
+    ----------
+    dim : int, optional
+        The dimension to reduce.
+
+    Returns
+    ----------
+    Tensor
+        Reduced tensor
+
+    Example
+    ----------
+
+    >>> row = torch.tensor([0, 1, 1])
+    >>> col = torch.tensor([0, 0, 2])
+    >>> val = torch.tensor([1, 1, 2])
+    >>> A = SparseMatrix(row, col, val, shape=(4, 3))
+    >>> print(A.smin())
+    tensor(1)
+    >>> print(A.smin(0))
+    tensor([1, 0, 2])
+    >>> print(A.smin(1))
+    tensor([1, 1, 0, 0])
+
+    >>> row = torch.tensor([0, 1, 1])
+    >>> col = torch.tensor([0, 0, 2])
+    >>> val = torch.tensor([[1, 2], [2, 1], [2, 2]])
+    >>> A = SparseMatrix(row, col, val, shape=(4, 3))
+    >>> print(A.min())
+    tensor([1, 1])
+    >>> print(A.min(0))
+    tensor([[1, 1], [0, 0], [2, 2]])
+    >>> print(A.min(1))
```

Comment:
I guess all these `min` should be `smin`.

Reply:
Right.